core/src/main/java/de/jplag/highlightextraction/WeightingStrategies.java

```diff
@@ -0,0 +1,28 @@
+package de.jplag.highlightextraction;
+
+/**
+ * This class contains the possible weighting functions for a match, in the frequency analysis.
+ */
+public enum WeightingStrategies {
```

Comment:
Merge the enum and the subclasses passed to the enums, please. No reason to have this duplication/separation. Still consider my naming proposal though.